🚀 2DMBIE 6.0: Extra Content

JoJ3o opened the pull request:
> SUMMARY
> ----------
> Online multiplayer on a blank scene, extra weapons and enemies, balancing weapon properties, critical hits and small features to increase user quality of life.

## Major features
- Special Wave system
- 1 new enemy type
- 7 new weapons
- Ammo pouch system
- Headshot System

## Other Features
- Options Cache
- highscore save file
- Jump buffers
- Storyline (Note system)
- Storyline (Cutscene)
- Parallex Background
- Game Over screen
- Perks QoL
- Small balancing: more ammo and bullet damage
- Debug mode
- Juke a Box (weapon upgrade machine)
- Short and Long slide

## Bug fixes
- Backfire bullet bug
- Save settings ingame
- Shotgun bullet fix
- Wave bar not going down when knifing fix
- Wave bar going down twice for 1 enemy fix
- Infinite slide bug
- Options menu fix
- Level 3 Repair
- Play sound signal fix
- Shotgun bullet penetration fix
- Bullet penetration fix
- Infinitie buying bug
- Door/weapon buying sound bug
- Zombie turning glitch
- Slide animation length bug
- Weapon switch sound bug

## Planned Major Features (2DMBIE 7.0)
- Implementeren Multiplayer????

#### Before merge

- [x] Tested to build, start and is usable
- [x] Changed code has been reviewed by 5 people
- [x] Empty debugger
- [x] No warning/errors
- [x] Complete Changelog
---

Commit:
bugged ammo pouch system

Commit:
semi fixed ammo pouch

Commit:
almost fixed ammo pouch

Commit:
offscreen marker

Commit:
bugged offscreen marker

Commit:
Moving marker on screen

Commit:
headshots

- headshot damage
- headshot score
- head removed (random)
- blood (when head removed)

Commit:
Merge branch 'alpha' into #118-Ammo-Pouches

Commit:
bugfix arrow invisible after pickup

Blamant committed:
UMP45 as purchasable weapon, scaling not fixed

Commit:
headshot

- headshot damage
- headshot score
- blood
- dissapearing head

JoJ3o committed:
Merge pull request #64 from 2DMBIE/#118-Ammo-Pouches

#118 ammo pouches

Commit:
Merge branch 'alpha' into #126-Headshot

JoJ3o committed:
Merge pull request #65 from 2DMBIE/#126-Headshot

#126 headshot

Commit:
HigScore system

highscore system
file save
file loading

TijmenSimons committed:
GameOver screen is finished

Added:
- Player does not dissappear when killed
- Instead of reloading the scene, the game over scene is displayed
  - Game over screen displays a message
  - Game over screen displays your score

Commit:
special wave system

- Special wave system
- (temporary) special wave enemy
- Headshot fix
- enemy speed fix
- prints gefixed

Commit:
headshot fix

TijmenSimons committed:
Gameover screen finished

Added wave display

TijmenSimons committed:
Merge branch 'main' into #133-GameOverScreen

TijmenSimons committed:
Fixed pausing while game over

Fixed pausing while game over

florisgravendeel committed:
Merge pull request #67 from 2DMBIE/#133-GameOverScreen

#133 game over screen

JoJ3o committed:
Rainbow gameover title pog

- Added game over sound
- Changed game over layout

JoJ3o committed:
Merge branch 'alpha' into #133-GameOverScreen

JoJ3o committed:
Reset game over screen

- reset game over screen
- debug reset during pause game fix
- remove print

ivardeviking committed:
Merge pull request #69 from 2DMBIE/#133-GameOverScreen

#133 game over screen

Commit:
Merge branch 'alpha' into #129-highscore-save

JoJ3o committed:
Game over screen + player death fix

JoJ3o committed:
Merge pull request #71 from 2DMBIE/#129-highscore-save

#129 highscore save

JoJ3o committed:
Debug mode - Optionsmenu Revamp

- Disable game reset
- Disable game debugger
- Add 25000 score/money

Blamant committed:
all new weapons added and audio bugs fixed

- M249
- M60
- XM1014
- P90
- Intervention
- AWP
- UMP 45

and audio bug sort of fixed

Commit:
highscore bug fix

TijmenSimons committed:
the perkening

All perks have sprites
Perks icon display on the interface
Buying a perk makes the interface perk visible
Perks also have according light glow color

TijmenSimons committed:
Merge pull request #73 from 2DMBIE/#134-Perks-Qol

the perkening

JoJ3o committed:
Rolling Zombie Heads

Commit:
Config file Setting

- Saves Audio Settings
- Saves Game options
- Makes/Updates Game Config
- Audio works

Commit:
Merge branch 'alpha' into #129-highscore-save

JoJ3o committed:
Merge pull request #74 from 2DMBIE/#129-highscore-save

#129 highscore save

Commit:
Merge branch 'alpha' into #130-GameOptionsConfig

TijmenSimons committed:
The Backlegrounds

not the battlegropunds or anything..

Added ground beneath the floor (building doesn't float anymore)
Add starry parallex background

JoJ3o committed:
Merge pull request #76 from 2DMBIE/#130-GameOptionsConfig

#130 game options config

TijmenSimons committed:
Merge branch 'alpha' into #121-parallex-background

JoJ3o committed:
Merge pull request #77 from 2DMBIE/#121-parallex-background

#121 parallex background

JoJ3o committed:
Merge branch 'alpha' into #135-DebugMode

Commit:
Merge branch 'alpha' into #94-Round/Wave-System

Commit:
fix special enemy, and speed

JoJ3o committed:
Merge pull request #78 from 2DMBIE/#94-Round/Wave-System

#94 round/wave system

JoJ3o committed:
Bug fixes

- Backfire bullet bug
- Save settings ingame
- Shotgun bullet fix

Commit:
"special enemy"

JoJ3o committed:
Merge branch 'alpha' into #135-DebugMode

JoJ3o committed:
Fix headshot heads

emit signal was at the wrong spot

ivardeviking committed:
Merge pull request #79 from 2DMBIE/#135-DebugMode

#135 debug mode

Blamant committed:
Audio bug fixes

Blamant committed:
Signal fix, decay fix, scale fix, faster shooting fix

Blamant committed:
Merge branch 'alpha' into #120-extra-weapons

Commit:
zombie healthbar

- added a zombie healthbar
- chanches colour depending on the amount of health
- isn't visible if enemy is 100% HP

Blamant committed:
reload perk fix and gun position fix

Commit:
new enemy sprite, scene, working special wave system

Commit:
Merge branch 'alpha' into #122-extra-enemies

Commit:
semi fixed enemy scene (sprite location animation bugged)

Commit:
special wave fix

Commit:
different variables for special wave enemies

Commit:
marker texture toegevoegd (pog pog)

JoJ3o committed:
Merge pull request #80 from 2DMBIE/#120-extra-weapons

#120 extra weapons

Commit:
dynamic ammo pouch fix

TijmenSimons committed:
Skellyboi animation fixed

Added skelleton animation

TijmenSimons committed:
specialwave fixed

forgot to undo setting the special wave to 1

Commit:
added jump buffers

TijmenSimons committed:
Merge branch 'alpha' into #122-extra-enemies

Commit:
jumpbuffers fully fixed

JoJ3o committed:
Wave bar not going down when knifing

Commit:
enemy collision box fix

TijmenSimons committed:
Zombie walk4 fixed

Zombie walk4 fixed

Commit:
z-index marker arrow fixed, rolling head enemy fixed

Commit:
reset special wave number, fix skeleton hitbox

JoJ3o committed:
Wavebar goes down twice for 1 enemy fix

JoJ3o committed:
signals ughh

need enemyhead node and zombie assigned to it

Commit:
falling throught the floor after being decapetated

TijmenSimons committed:
Part 1

Added explosion part of intro cutscene

TijmenSimons committed:
Commit for safety

:)

TijmenSimons committed:
Safety Commit

Commit:
added notes

- added not sprite
- added story array
- added popup node

Blamant committed:
Working PAP

TijmenSimons committed:
Part 2, zombies are coming

In progress ._.

Blamant committed:
m249 m4a1 and mp5 pap'ed

Commit:
random spawn note + interactable note

Commit:
fixed note spawn

Commit:
Note popup fix,

JoJ3o committed:
Set texture of rolling enemy head to whatever enemy it came from

JoJ3o committed:
Changed rolling head chance back to 15%

ivardeviking committed:
Merge pull request #81 from 2DMBIE/#122-extra-enemies

#122 extra enemies

Commit:
Merge branch 'alpha' into #Hotfix-Branch

JoJ3o committed:
Merge pull request #84 from 2DMBIE/#Hotfix-Branch

#hotfix branch

Blamant committed:
new sounds added and bullet sprites

Commit:
note pause system, note text

TijmenSimons committed:
Finished Cutscene

TijmenSimons committed:
Merge branch '#128-Story/Intro-Cutscene' of https://github.com/2DMBIE/2DMBIE into #128-Story/Intro-Cutscene

TijmenSimons committed:
Merge branch 'alpha' into #128-Story/Intro-Cutscene

TijmenSimons committed:
Small bug fix

Fixed not being able to pause after picking up note

Blamant committed:
All weapons sounds, sprites, bullets added. Also Jukebox added. Also bug fixes and buy/NEM sounds added

JoJ3o committed:
Merge pull request #86 from 2DMBIE/#128-Story/Intro-Cutscene

#128 story/intro cutscene

Blamant committed:
Merge branch 'alpha' into #128-puck-a-panch

Blamant committed:
Electrical humming added and bugs fixed

Commit:
script path fix

Blamant committed:
Multiple bug fixes for pull request

- Fixed sprite size
- Adjusted interface to accommodate for large weapon sprites
- Added broken light to the Jukebox
- Added light2d to the laser bullets
- Fixed humming sound not looping

Commit:
fix path (again)

florisgravendeel commented on 2DMBIE/assets/scenes/stickyNote.tscn:
```diff
@@ -0,0 +1,31 @@
+texture = ExtResource( 1 )
+
+[node name="Light2D" type="Light2D" parent="."]
+texture = ExtResource( 2 )
```
@JoJ3o multiplayer breekt singleplayer, we kunnen pas mergen als we verder zijn.

JoJ3o committed:
Fix level 3

JoJ3o committed:
Merge branch '#Hotfix-Branch' of https://github.com/2DMBIE/2DMBIE into #Hotfix-Branch

JoJ3o committed:
Implement cutscene to start of the game

JoJ3o committed:
Change perk and pap positions

TijmenSimons committed:
Button Inputs Shown

Button Inputs Shown on controls optionmenu

JoJ3o committed:
Little Improvements

- Move tombstone behind door
- Give healthbar a number
- Reorder weapon slots
- Give weapon slots a number
- Replace weapons and perks
- Reduce regen time

JoJ3o committed:
Merge pull request #88 from 2DMBIE/#128-puck-a-panch

🚀 #128 puck a panch

JoJ3o committed:
Merge branch 'alpha' into #Hotfix-Branch

JoJ3o committed:
Move movement perk

TijmenSimons committed:
Added transition scene

Added transition scene

Commit:
Merge branch 'alpha' into #138-DarkenedAreas

TijmenSimons committed:
Tip Screen

Added tip screen when loading the map.

TijmenSimons committed:
Merge branch 'alpha' into US140-Controls-Display

TijmenSimons committed:
Tested the stuff

Found a small bug, but it has been fixed in another branch already (says Joas)

Blamant committed:
MovementPerk with sliding bug fixed

Blamant committed:
White lines fix Use Pixel Snap on

TijmenSimons committed:
Added kills to gameover

Added kills to gameover
Fixed spelling error in the transition tooltip text

TijmenSimons committed:
Merge pull request #91 from 2DMBIE/#Hotfix-Branch

#hotfix branch

TijmenSimons committed:
Merge branch 'alpha' into #142-Aantal-Enemy-Kills

TijmenSimons committed:
commit

TijmenSimons committed:
Merge branch 'alpha' into US140-Controls-Display

TijmenSimons committed:
Merge pull request #93 from 2DMBIE/US140-Controls-Display

#140-controls-display

TijmenSimons committed:
Merge branch 'alpha' into #142-Aantal-Enemy-Kills

Commit:
darkened areas

- added dark areas
- light up when unlocked

TijmenSimons committed:
Small fix and bug find

Fixed perk price text, wasn't visible on a white background.

Commit:
Merge branch 'alpha' into #138-DarkenedAreas

TijmenSimons committed:
Small changes

Had a bug where HP is weird (e.g. i had 1224 max hp displayed), could find the source, made the hp maxhealth when above maxheatlh

Blamant committed:
Weapon balance changes

Changed all values, see the Google Spreadsheet for the exact values

Blamant committed:
Mystery box added V1

TijmenSimons committed:
Fixes some pains in the ass bugs

Fixes some pains in the ass bugs
Dying on specialwave caused the special wave to always be active
A weird error which was fixed by dis- and reconnecting a signal.

TijmenSimons committed:
Merge pull request #94 from 2DMBIE/#138-DarkenedAreas

#138 darkened areas

Blamant committed:
Mystery box added in the middle of the map

TijmenSimons committed:
Merge branch 'alpha' into #142-Aantal-Enemy-Kills

Commit:
Cursor script

Commit:
Cursor scene

Commit:
custom cursor

Blamant committed:
Mystery box fixed

Mystery box now works with the inventory system

Blamant committed:
Lights position fixed library

TijmenSimons committed:
Bug fixes

fixed:
Bullet-Player collision
maxhealth now ectually is the max health
zombies do less damage
Bullet gets paused on pause
game values reset when pressing exit buttons

TijmenSimons committed:
Merge pull request #95 from 2DMBIE/#Hotfix-Branch

#hotfix branch

TijmenSimons committed:
Merge branch 'alpha' into #142-Aantal-Enemy-Kills

ivardeviking committed:
Merge pull request #96 from 2DMBIE/#142-Aantal-Enemy-Kills

#142 aantal enemy kills

Blamant committed:
Knife kill gives 200 score

Blamant committed:
Signals fixed level 2 perks

Blamant committed:
Penetration of your mom shotguns

Blamant committed:
Movement debugger deleted

TijmenSimons committed:
Added More Music

Made music folder, moved the music fileds, repaired dependencies.

Blamant committed:
Balance changes

-MP5JaB changed bullet penetration to 2
- LMG'S AMMO REDUCED
- Fullscreen

Blamant committed:
spritesheets added and arcade machine music

TijmenSimons committed:
Merge pull request #97 from 2DMBIE/#136-MysteryBox

#136 mystery box

TijmenSimons committed:
Merge branch 'alpha' into #144---Diverse-Theme-Songs

TijmenSimons committed:
Fixed a merge error

While merging, level.tscn died.
Repaired it and readded the music features.
Also changed gameover music to the menu music. Woodwind gets annoying.

TijmenSimons committed:
Added time played to gameover screen

Added time played to gameover screen

ivardeviking committed:
Merge pull request #98 from 2DMBIE/#144---Diverse-Theme-Songs

#144 diverse theme songs

ivardeviking reviewed (approved):
De alpha werk, dus het is wel best
